gvsbuild/build.py

```diff
@@ -0,0 +1,437 @@
+        None,
+        help="Command line options to pass to msbuild.",
+        rich_help_panel="Options to Pass to Build Systems",
+    ),
+    skip: list[str] = typer.Option(
+        None,
+        help="List of projects to skip. i.e gtk3, glib, ...",
+        rich_help_panel="Skip and Cleanup Options",
+    ),
+    use_env: bool = typer.Option(
+        False,
+        help="Use and keep the calling environment for LIB, LIBPATH, INCLUDE and PATH",
+        rich_help_panel="Environment Options",
+    ),
+    make_zip: bool = typer.Option(
+        False,
+        help="Create singles zips of the projects built under the build-dir\\dist\\vsXXXX[-sdkVer]\\[platform]\\[configuration], "
+        "for example 'C:\\gtk-build\\dist\\vs2015-8.1\\win32\\release'. "
+        "NOTE: the destination dir (e.g. 'C:\\gtk-build\\gtk\\win32\\release') "
+        "will be cleared completely before the build!",
+        rich_help_panel="Zip Options",
+    ),
+    zip_continue: bool = typer.Option(
+        False,
+        help="If set, don't initialize the zip creation phase and keep the destination dir.",
+        rich_help_panel="Zip Options",
+    ),
+    from_scratch: bool = typer.Option(
+        False,
+        help="Start from scratch, deleting the build and the "
+        "destination directories of the project for the current "
+        "platform/configuration "
+        "setup (e.g. 'C:\\gtk-build\\build\\x64\\release' and "
+        "'C:\\gtk-build\\gtk\\x64\\release' and the common tools ('C:\\gtk-build\\tools')",
+        rich_help_panel="Skip and Cleanup Options",
+    ),
+    keep_tools: bool = typer.Option(
+        False,
+        help="Active only when used with --from-scratch, keep and don't delete the (common) tool directory.",
+        rich_help_panel="Skip and Cleanup Options",
+    ),
+    fast_build: bool = typer.Option(
+        False,
+        help="Don't build a project if it's already built and not updated."
+        "Note: you can have wrong results if you change only the patches or the script (updating the tarball or "
+        "the git source is handled correctly)",
+        rich_help_panel="Skip and Cleanup Options",
+    ),
+    keep_going: bool = typer.Option(
+        False,
+        help="Continue the build even on errors, dropping the projects that depends on the failed ones",
+        rich_help_panel="Skip and Cleanup Options",
+    ),
+    py_wheel: bool = typer.Option(
+        False,
+        help="pycairo/pygobject: build also the wheel distribution format",
+        rich_help_panel="Introspection Options",
+    ),
+    enable_gi: bool = typer.Option(
+        False,
+        help="For the GTK stack, create the .gir/.typelib files for gobject introspection",
+        rich_help_panel="Introspection Options",
+    ),
+    ffmpeg_enable_gpl: bool = typer.Option(
+        False,
+        help="ffmpeg: build with the gpl libraries/modules",
+        rich_help_panel="FFmpeg Options",
+    ),
```

Comment:
note: at some point this should go to some type of project configuration options

Reply:
Ok, but not part of this PR.